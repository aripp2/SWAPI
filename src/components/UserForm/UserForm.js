import React, { Component } from 'react';
import './UserForm.scss';

class UserForm extends Component {
  constructor() {
    super();
    this.state = {

    }
  }

  render() {
    return (
      <form>This will be the form</form>
      )
  }
}

export default UserForm;