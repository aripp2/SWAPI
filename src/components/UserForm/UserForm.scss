@import './src/variables.scss';

.user_profile-div {
  display:flex; 
  justify-content: flex-end;
  flex-direction: column-reverse;
  height: 100vh;
  width: 100%;
  background-color: #F0FFFF;
  position:relative;
}

.user_form {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  margin: 3% 8% 2% 5%;
  position: absolute;
  top: 3%;
  right: 2%;
  width: 43%;
  box-shadow: 0px 5px 5px 0px #e8e8e8, 0px 10px 10px 0px #e8e8e8;
  height: 85%;
  background: white;

}
.user_radio-div {
  display:flex;
  flex-direction: row;
}
.user_radio-btn-div{
  position:relative;
  top: 115px;
  left: -11px;
}
.user_radio-input {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  position: relative;
  background-color: #f1f1f1;
  color: #666;
  top: 10px;
  height: 30px;
  width: 30px;
  border: 0;
  border-radius: 50px;
  cursor: pointer;     
  margin-right: 7px;
  outline: none;
}

.user_radio-input:checked::before {
  position: absolute;
  font: 13px/1 'Open Sans', sans-serif;
  left: 11px;
  top: 7px;
  content: '\02143';
  transform: rotate(40deg);
}
.user_radio-input:hover
{
  background-color: #f7f7f7;
}
.user_radio-input:checked
{
  background-color: #f1f1f1;
}
.user_radio-label
{
  font: 15px/1.7 'Open Sans', sans-serif;
  color: #333;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  cursor: pointer;
} 
.user_profile-h1-div { 
  position: absolute;
  top: 0px;
  width: 100%;
  height: 50%;
  background-color: white;
  margin: 0;
}

.user_profile-h1 { 
  position: absolute;
  top: 38%;
  left: 9%;
  background-color: white;
  margin: 0;
  font-size: 50px;
  text-shadow: 2px 2px 2px #dfdfdf;
}

.user-submit-btn {
  font-size: large;
  background-color: #f0ffffba;
  font-weight: 700;
  letter-spacing: 1.1px;
  border: .5px solid #80808021;
  width: 100px;
  box-shadow: 0px 1px 1px 0px #e8e8e8, 0px 3px 3px 0px #e8e8e8;
}


.user_quote-textarea {
  border: 0;
  outline: 0;
  background: transparent;
  border-bottom: 1px solid black;
}

.user_radio-title {
  position: absolute;
  right: 24px;
  top: -60px;
  width: 217px;
  font-weight: 700;
}

.user_quote-div, 
.user_name-div {
  position: relative;   
}

.user_quote-label {
  position: absolute;
  bottom: -12px;
  right: -30px;
  width: 160px;
  font-weight: 700;
}

.user_name-label {
  position: absolute;
  bottom: -20px;
  right: 84px;
  font-weight: 700;
}

.user_quote-textarea,
.user_name-input {
  border: 0;
  outline: 0;
  background: transparent;
  border-bottom: 1px solid black;
  width: 262px;
  position: absolute;
  top: 33px;
  right: -140px; 
}


.user_profile-image {
  position: absolute;
  bottom: 33px;
  left: 8%;
  height: 40%;
}
