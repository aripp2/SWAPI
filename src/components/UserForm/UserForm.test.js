import React from 'react'; 
import { shallow } from 'enzyme';
import UserForm from './UserForm';

describe('UserForm', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<UserForm 
      user={
        ({ name: 'kate',
        quote: 'hello', 
        knowledgeLevel: 'Star What?'
      })} 
      submitUser={jest.fn()}
    />)
    expect(wrapper).toMatchSnapshot(); 
  })
  
  it('should set state', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(<UserForm 
      submitUser={mockFunction}
    />)
    wrapper.instance().handleChange = jest.fn()
    wrapper.find('input').at(0).simulate('change')
    expect(wrapper.instance().handleChange).toHaveBeenCalled()
  })

  it('should set state for textarea', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(<UserForm 
      submitUser={mockFunction}
    />)
    wrapper.instance().handleChange = jest.fn()
    wrapper.find('textarea').simulate('change')
    expect(wrapper.instance().handleChange).toHaveBeenCalled()
    })

  it('should set state for radio button at index 1', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(<UserForm 
      submitUser={mockFunction}
    />)
    wrapper.instance().handleChange = jest.fn()
    wrapper.find('input').at(1).simulate('change')
    expect(wrapper.instance().handleChange).toHaveBeenCalled()
  })


  it('should set state for radio button at index 1', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(<UserForm 
      submitUser={mockFunction}
    />)
    wrapper.instance().handleChange = jest.fn()
    wrapper.find('input').at(2).simulate('change')
    expect(wrapper.instance().handleChange).toHaveBeenCalled()
  })


  it('should set state for radio button at index 1', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(<UserForm 
      submitUser={mockFunction}
    />)
    wrapper.instance().handleChange = jest.fn()
    wrapper.find('input').at(3).simulate('change')
    expect(wrapper.instance().handleChange).toHaveBeenCalled()
  })

  it('should change state based on an event', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(<UserForm 
      submitUser={mockFunction}
    />)
    const mockEvent = {
      target: {
        name: 'name', value: 'Kate' 
      } 
    }

    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('name')).toEqual('Kate')
  })

    it('should change state based on an event', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(<UserForm 
      submitUser={mockFunction}
    />)
    const mockEvent = {
      target: {
        name: 'quote', value: 'pants' 
      } 
    }

    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('quote')).toEqual('pants')
  })

    it('should change state based on an event', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(<UserForm 
      submitUser={mockFunction}
    />)
    const mockEvent = {
      target: {
        name: 'knowledgeLevel', value: 'one' 
      } 
    }

    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('knowledgeLevel')).toEqual('one')
  })

  it('should call submit user when handleFormSubmit is called', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(<UserForm 
      submitUser={mockFunction}
    />)

    wrapper.instance().handleFormSubmit()
    expect(mockFunction).toHaveBeenCalled()
    
  })

})