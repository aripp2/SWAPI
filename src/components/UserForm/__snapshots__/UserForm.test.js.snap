// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm should match snapshot 1`] = `
<div
  className="user_profile-div"
>
  <img
    alt="Yo Jedi laser gif animation gif r2d2 cat menu starwars force jedi"
    className="user_profile-image"
    src="https://cdn.dribbble.com/users/1539273/screenshots/6122439/spoilers_drib.gif"
  />
  <div
    className="user_profile-h1-div"
  >
    <h1
      className="user_profile-h1"
    >
      WELCOME TO 
      <br />
       SWAPI TRIVIA! 
    </h1>
  </div>
  <form
    className="user_form"
  >
    <div
      className="user_name-div"
    >
      <label
        className="user_name-label"
      >
        Name
      </label>
      <input
        className="user_name-input"
        name="name"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="user_quote-div"
    >
      <label
        className="user_quote-label"
      >
        Favorite SW Quote?
      </label>
      <textarea
        className="user_quote-textarea"
        name="quote"
        onChange={[Function]}
        value=""
      />
    </div>
    <div
      className="user_radio-btn-div"
    >
      <label
        className="user_radio-title"
      >
        How much do you know about Star Wars?
      </label>
      <div
        className="user_radio-div"
      >
        <label
          className="user_radio-label"
        >
          <input
            className="user_radio-input"
            name="knowledgeLevel"
            onChange={[Function]}
            type="radio"
            value="Star What?"
          />
          Star What?
        </label>
      </div>
      <div
        className="user_radio-div"
      >
        <label
          className="user_radio-label"
        >
          <input
            className="user_radio-input"
            name="knowledgeLevel"
            onChange={[Function]}
            type="radio"
            value="Jedi in Training?"
          />
          Jedi in Training
        </label>
      </div>
      <div
        className="user_radio-div"
      >
        <label
          className="user_radio-label"
        >
          <input
            className="user_radio-input"
            name="knowledgeLevel"
            onChange={[Function]}
            type="radio"
            value="My firtsborns name is yoda"
          />
          My Firstborns names is Yoda
        </label>
      </div>
    </div>
    <Link
      to="/movies"
    >
      <button
        className="user-submit-btn"
        disabled={true}
        onClick={[Function]}
        type="submit"
      >
        ENTER
      </button>
    </Link>
  </form>
</div>
`;
